contracts/governance/TimelockController.sol

```diff
@@ -31,6 +32,43 @@ contract TimelockController is AccessControl, IERC721Receiver, IERC1155Receiver
     mapping(bytes32 => uint256) private _timestamps;
     uint256 private _minDelay;
 
+    enum OperationState {
+        Unset,
+        Pending,
+        Ready,
+        Done
+    }
+
+    /**
+     * @dev Mismatch between the parameters length for a operation call.
+     */
+    error TimelockInvalidOperationLength(uint256 targets, uint256 payloads, uint256 values);
+
+    /**
+     * @dev The schedule operation doesn't met the minimum delay.
+     */
+    error TimelockInsufficientDelay(uint256 delay, uint256 minDelay);
+
+    /**
+     * @dev The current state of an operation is not the required.
+     */
+    error TimelockIncorrectState(bytes32 operationId, OperationState expected);
+
+    /**
+     * @dev A call to a target failed. The target may have reverted.
+     */
+    error TimelockFailedCall();
+
+    /**
+     * @dev The predecessor to an operation not yet done.
+     */
+    error TimelockMissingPredecessor(bytes32 predecessorId);
```

Comment:
`Missing` -> `Pending` ?

"Missing" sounds like "unknown" to me.

Reply:
IMO `Pending` would mean "we check the state and its pending" when the truth is "its not executed" (with no check that the corresponding task is even scheduled).

Reply:
`TimelockUnsettledPredecessor` ?

Reply:
I'm going for `TimelockUnexecutedPredecessor` since I think it's a better description, but let me know if you have strong opinions for `Pending` or `Unsettled`